lib/spack/spack/spec.py

```diff
@@ -299,22 +301,29 @@ def _autospec(self, spec_like):
         return ArchSpec(spec_like)
 
     def _cmp_iter(self):
-        yield self.platform
+        yield self._platforms
         yield self.os
         yield self.target
 
     @property
     def platform(self):
         """The platform of the architecture."""
-        return self._platform
+        if len(self._platforms) == 1:
+            return self._platforms[0]
+        if len(self._platforms) == 0:
+            return None
+        raise ValueError(f"Platform not concrete: {self._platforms}")
```

Comment:
Do we care that this is not consistent?  It seems like it should raise (as `.versions` does) when `len` is `0` as well.  No?

Reply:
Hm, changing that behavior causes 78 test failures, seems like it's relied on.